keps/sig-node/3327-align-by-socket/README.md

```diff
@@ -0,0 +1,252 @@
+
+### Graduation Criteria
+
+#### Alpha
+
+- [X] Implement the new policy option.
+- [X] Ensure proper unit tests are in place.
+- [X] Ensure proper e2e node tests are in place.
+
+#### Beta
+
+- [X] Gather feedback from consumers of the new policy option.
+- [X] Verify no major bugs reported in the previous cycle.
+
+#### GA
+
+- [X] Allow time for feedback (1 year).
+- [X] Make sure all risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+We expect no impact. The new policy option is opt-in and orthogonal to the existing ones.
+
+### Version Skew Strategy
+
+No changes needed
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyAlphaOptions`
+  - Components depending on the feature gate: `kubelet`
+- [X] Change the kubelet configuration to set a `CPUManager` policy of `static` and a `CPUManager` policy option of `align-by-socket`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+	Yes -- a kubelet restart is required.
+
+###### Does enabling the feature change any default behavior?
+
+No. In order to trigger any of the new logic, three things have to be true:
+1. The `CPUManagerPolicyOptions` feature gate must be enabled
+1. The `static` `CPUManager` policy must be selected
+1. The new `align-by-socket` policy option must be selected
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either:
+1. Disabling the `CPUManagerPolicyOptions` feature gate
+1. Switching the `CPUManager` policy to `none`
+1. Removing `align-by-socket` from the list of `CPUManager` policy options
+
+Existing workloads will continue to run uninterrupted, with any future workloads having their CPUs allocated according to the policy in place after the rollback.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+No changes. Existing container will not see their allocation changed. New containers will.
+
+###### Are there any tests for feature enablement/disablement?
+
+- A specific e2e test will demonstrate that the default behaviour is preserved when the feature gate is disabled, or when the feature is not used (2 separate tests)
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Inspect the kubelet configuration of a node -- check for the presence of the feature gate and usage of the new policy option.
+
+###### How can someone using this feature know that it is working for their instance?
```

Comment:
AFAIK, but I'll obviously defer the autorithative answer to PRRs, that this is meant to ask for more readily observable options, like metrics, or object statuses, or system properties. This (detailed) tests works of course.

Reply:
Thanks for pointing out! There was not no relevant metric for CPUManager policy option, so put down the test case taking a cue from other policy options KEP.